client/post-editor/media-modal/detail/detail-preview-videopress.jsx

```diff
@@ -69,6 +70,44 @@ class EditorMediaModalDetailPreviewVideoPress extends Component {
 		return false;
 	}
 
+	// Run requests through the rest proxy to support loading private videos.
+	requestProvider = ( observable ) => {
+		return () => {
+			return observable( ( set, reject, done ) => {
+				const { videopress_guid: guid } = this.props.item;
+
+				if ( ! guid ) {
+					return;
+				}
+
+				const getVideo = ( id ) => {
+					const data = { path: '/videos/' + id };
+
+					wpcomProxyRequest( data, function ( err, body, headers ) {
+						if ( err ) {
+							return;
+						}
+
+						// If an upload_date property is present, we have a valid response.
+						if ( body.upload_date != null ) {
+							done( body );
+						} else {
+							const error = new Error( body ? body.message : 'Unknown' );
+							error.code = body ? body.error : null;
+							error.errorMessage = body ? body.errorMessage : null;
+							error.status = headers.status;
+							reject( error );
+						}
+					} );
+				};
+
+				getVideo( guid );
+
+				return () => {};
```

Comment:
this is a function that is called to cancel this observable. If that wpcomProxyRequest is cancellable, this is the place to put it

Reply:
Thanks, I didn't see any cancel functionality in it though.